### PR TITLE
 bug/2403 scheduled triggers

### LINKED ISSUE #2403: [Bug] Investigate scheduled triggers for nightly scan
**Description:**

Since v3.0.0 was merged into HHS/TANF-app main and master branches, the [scheduled trigger](https://github.com/raft-tech/TANF-app/pull/2284) for the nightly OWASP scan is triggering all the workflows instead of just the nightly scan (see below). 

The [recent history and insights from CircleCI for raft-tech/TANF-app](https://app.circleci.com/insights/github/raft-tech/TANF-app/workflows/build-and-test/overview?branch=develop&reporting-window=last-90-days) suggest that all workflows (except those recently implemented --eg. `enable-versioning-for-s3-buckets`) have been running daily for at least the last 90 days. 

Until this is resolved, the scheduled triggers intended for the nightly scan on the HHS side have been disabled. 

_develop (raft-tech)_
Screenshot 2023-02-08 at 8 04 54 PM

_main and master (HHS)_

Screenshot 2023-02-08 at 8 11 23 PM

**Acceptance Criteria:** 
- [ ] Nightly workflow triggers at 12AM UTC, daily
- [ ] no other workflows trigger at 12AM UTC, daily
- [ ] Testing Checklist has been run and all tests pass
- [ ] README is updated, if necessary

**Tasks:**
- [ ] _troubleshoot_
- [ ] _update circleci config_
- [ ] Run Testing Checklist and confirm all tests pass

**Supporting Documentation:**
- [about scheduled workflows with dynamic config](https://support.circleci.com/hc/en-us/articles/360060833331-Support-for-Scheduled-Workflows-in-Dynamic-Configurations-Setup-Workflows)
- [workaround ](https://discuss.circleci.com/t/workaround-using-scheduled-workflows-with-dynamic-config/40344)

### PR DESCRIPTION
**Pull request by n0remac**
## Summary of Changes
_Provide a brief summary of changes_
Pull request closes #2403 

Just needed to add an unless keyword to the workflows that we do not want to trigger.

This was tested on by changing the branch name in the config to run from this branch and by updating the schedule to run every five minutes.

Screen Shot 2023-02-15 at 12 13 30 PM

**Acceptance Criteria:** 
- [x] Nightly workflow triggers at 12AM UTC, daily
- [x] no other workflows trigger at 12AM UTC, daily
- [x] Testing Checklist has been run and all tests pass
- [x] README is updated, if necessary

**Tasks:**
- [x] _troubleshoot_
- [x] _update circleci config_
- [x] Run Testing Checklist and confirm all tests pass

### CODE REVIEW

**Review by andrew-jameson — COMMENTED** [review]
Why do we have to have this flag on all these workflows? Could we only have this on the relevant owasp scan job? I'm fine to approve but it seems a bit off

**Review by George-Hudson — APPROVED** [review]

**Comment by George-Hudson** [comment]
> Why do we have to have this flag on all these workflows? Could we only have this on the relevant owasp scan job? I'm fine to approve but it seems a bit off

It would be cleaner eventually to trigger each workflow `when` events we want to trigger them happen instead of having a growing list of `unless`, but that is a bit heavier lift and I'll be looking at that when we look into refactoring the CI/CD pipeline, if that's okay, @andrew-jameson 

Understood, will approve for now and we can pay tech debt later. Can you write up a stub/draft ticket explaining this and tag this ticket 2414 in it? Please and thank you. @George-Hudson

**Review by andrew-jameson — APPROVED** [review]

File: .circleci/deployment/workflows.yml [hunk 1]
```diff
@@ -11,7 +11,10 @@
             - deploy-infrastructure-dev
 
   staging-deployment:
```
Comment (ADPennington) [inline .circleci/deployment/workflows.yml:13]: @n0remac @George-Hudson does this also prevent all the workflows from running on main (hhs staging) and master (hhs production)?

File: .circleci/deployment/workflows.yml [hunk 2]
```diff
@@ -51,7 +54,10 @@
                 - develop
 
   production-deployment:
-    unless: << pipeline.parameters.run_dev_deployment >>
+    unless:
+      or:
+        - << pipeline.parameters.run_dev_deployment >>
+        - << pipeline.parameters.run_nightly_owasp_scan >>
     jobs:
```
Comment (George-Hudson) [inline .circleci/deployment/workflows.yml:61]: @ADPennington added this logic in for production-deployment now too. Good catch!

**Review by ADPennington — APPROVED** [review]
thanks @George-Hudson @n0remac 👍🏾

**Pull request merged**